src/neo/Ledger/MemoryPool.cs

```diff
@@ -258,21 +259,28 @@ internal bool CanTransactionFitInPool(Transaction tx)
         /// Note: This must only be called from a single thread (the Blockchain actor). To add a transaction to the pool
         ///       tell the Blockchain actor about the transaction.
         /// </summary>
+        /// <param name="snapshot">Snapshot</param>
         /// <param name="hash"></param>
         /// <param name="tx"></param>
         /// <returns></returns>
-        internal bool TryAdd(UInt256 hash, Transaction tx)
+        internal bool TryAdd(SnapshotView snapshot, UInt256 hash, Transaction tx)
         {
             var poolItem = new PoolItem(tx);
 
+            _txRwLock.EnterWriteLock();
             if (_unsortedTransactions.ContainsKey(hash)) return false;
 
             List<Transaction> removedTransactions = null;
-            _txRwLock.EnterWriteLock();
             try
             {
+                // Check balance
+                BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
                 _unsortedTransactions.Add(hash, poolItem);
-                SendersFeeMonitor.AddSenderFee(tx);
+                if (balance < SendersFeeMonitor.AddSenderFee(tx))
+                {
+                    _unsortedTransactions.Remove(hash);
```

Comment:
Should add `SendersFeeMonitor.RemoveSenderFee(tx)` ?

Reply:
Yes, I will create a new method, knowing the balance

Reply:
Please, check it again